PersonaStats.Rmd

```diff
@@ -0,0 +1,287 @@
+---
+title: "Personas"
+author: "Kate Isaac"
+date: "`r Sys.Date()`"
+output: html_document
+---
+
+```{r}
+library(tidyverse)
+library(dplyr)
+library(data.table)
+library(magrittr)
+library(here)
+```
+
+```{r}
+resultsTidy <- readRDS(here("data/resultsTidy.rds"))
+```
+
+```{r}
+kindOfWorkDF <- resultsTidy %>%
+  separate(KindOfWork,
+           c("whichWorkA", "whichWorkB", "whichWorkC", "whichWorkD", "whichWorkE", "whichWorkF", "whichWorkG", "whichWorkH", "whichWorkI", "whichWorkJ"),
+           sep=", ", fill="right") %>%
+  pivot_longer(starts_with("whichWork"), 
+               values_to = "whichWorkDescription") %>%
+  dplyr::select(Timestamp, UserType, whichWorkDescription) %>%
+  mutate(whichWorkDescription =
+           recode(whichWorkDescription,
+                  "computational education" = "Computational education",
+                  "Program administration," = "Program administration"),
+         whichWorkDescription =
+           factor(whichWorkDescription),
+         Timestamp = 
+           factor(Timestamp)
+         ) %>%
+  drop_na() %>% 
+  pivot_wider(id_cols = Timestamp,
+              names_from = whichWorkDescription,
+              values_from = whichWorkDescription)
+```
+
+## Number of reported kinds of work description for each response
+
+add a column with the count of number of descriptions for each respondent and then use the table function to see how many responses gave 1, 2, 3, ... descriptions for their kind of work
+
+```{r}
+kindOfWorkDF %<>% mutate(count_na = rowSums(is.na(.)),
+                          num_descriptions = ncol(kindOfWorkDF) - 1 - count_na)
+
+numSelected <- table(kindOfWorkDF$num_descriptions) %>% 
+  as.data.frame() %>%
+  mutate(percentage_responses = round(Freq/nrow(resultsTidy)*100, 0))
+numSelected
```

Comment:
Another suggestion.. the boolean in the cells makes it a bit easier to count
```
#From above
kindOfWorkDF <- 
  resultsTidy %>% 
  select(Timestamp, KindOfWork) %>% 
  separate_longer_delim(KindOfWork, delim = ", ") %>% 
  mutate(value = TRUE) %>% 
  pivot_wider(names_from = KindOfWork, values_from = value, values_fill = FALSE)

numSelected <- kindOfWorkDF %>% select(!Timestamp) %>% rowSums()
tibble("X" = numSelected) %>% count(X)
```

Reply:
I agree -- I love the booleans for counting!